source/common/http/alternate_protocols_cache_impl.cc

```diff
@@ -174,38 +169,55 @@ std::chrono::microseconds AlternateProtocolsCacheImpl::getSrtt(const Origin& ori
   return entry_it->second.srtt;
 }
 
-void AlternateProtocolsCacheImpl::setAlternativesImpl(const Origin& origin,
-                                                      std::vector<AlternateProtocol>& protocols) {
-  static const size_t max_protocols = 10;
-  if (protocols.size() > max_protocols) {
-    ENVOY_LOG_MISC(trace, "Too many alternate protocols: {}, truncating", protocols.size());
-    protocols.erase(protocols.begin() + max_protocols, protocols.end());
+AlternateProtocolsCacheImpl::ProtocolsMap::iterator
+AlternateProtocolsCacheImpl::setAlternativesImpl(const Origin& origin,
+                                                 OptRef<std::vector<AlternateProtocol>> protocols,
+                                                 std::chrono::microseconds srtt,
+                                                 Http3StatusTrackerPtr&& tracker) {
```

Comment:
I'm wondering if it's worth it to split this into 3 different methods for alt protocols, srtt and tracker respectively. If we are adding more properties to OriginData, we won't want to change this interface all the time.

Reply:
hm, I don't want one function per as there's going to be combinations.  how about a struct, so we don't have to change each call site as we add more optionals?

Reply:
a struct to wrap all of them sounds good